File: orders-service/src/main/java/org/example/ordersservice/models/Producto.java
package org.example.ordersservice.models;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Table(name = "producto")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 45)
    private String nombre;

    private String foto;

    @Column(length = 500)
    private String descripcion;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal precio;

    @Column(nullable = false)
    private Integer cantidad;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "empresa_id", nullable = false)
    private Empresa empresa;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof User)) return false;
        User user = (User) o;
        return id != null && id.equals(user.getId());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: front/src/app/types/producto.ts
export interface Producto {
    id?: number;
    nombre: string;
    foto?: string;
    descripcion: string;
    precio: number;
    cantidad: number;
    empresaId?: number;
}

export interface ProductoInputDto {
    nombre: string;
    foto?: string;
    descripcion: string;
    precio: number;
    cantidad: number;
    empresaId: number;
}

export interface ProductoOutputDto {
    id: number;
    nombre: string;
    foto?: string;
    descripcion: string;
    precio: number;
    cantidad: number;
    empresaId: number;
    nombreEmpresa: string;
}